sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -1360,6 +1361,25 @@ private Function rewriteAndValidateFields(Function function, Context context) {
             return function;
         }
 
+        /**
+         * Converts the Symbol value(s) to Lucene string(s) to match against the Lucene ids.
+         * @param symbol The Symbol to convert to a String uid.
+         * @return A literal holding the Lucene query value string.
+         */
+        private static Literal generateLuceneUIDs(Symbol symbol) {
+            if (DataTypes.isCollectionType(symbol.valueType())) {
+                List<String> strings = ValueSymbolVisitor.STRING_LIST.process(symbol);
```

Comment:
ValueSymbolVisitor is deprecated

Reply:
If it's deprecated, it should be annotated as such. Where do you see deprecation?

Reply:
https://github.com/crate/crate/blob/a4dd6bd56f7ec37a4d2f13c553825abc87d45feb/sql/src/main/java/io/crate/analyze/symbol/ValueSymbolVisitor.java#L89-L105

Reply:
Deprecation on a static field. I'd propagate the deprecated to the actual class. Or wait, is it just deprecated for string lists? why?

Reply:
it's for all public methods

Reply:
I don't see any public methods inside `ValueSymbolVisitor` which are marked deprecated.

Reply:
Ah, it **just** got deprecated. I'm based on the branch were it isn't. Would have come up during rebasing then.

Reply:
https://github.com/crate/crate/pull/6576

Reply:
I think it would be okay to keep using the deprecated functionality for this PR - given that this is a fix and needs to be backported as well.

It will take a while until we can get rid of the `ValueSymbolVisitor` completely.
It's important that we don't introduce new features using the deprecated functionality, but I think for bugfixes it's fine.

Reply:
I think this is a fair point. Actually not using the ValueSymbolVisitor also requires a bit more work (porting the existing functionality to the new base class); and this should be done also for other instances once we remove the ValueSymbolVisitor.

If you're ok with that, I'd keep it as-is, @chaudum.

Reply:
@mxm ok for me also

Reply:
@chaudum Could you approve then?